npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -42,9 +43,31 @@ export const Cypress = (
   const indexHtmlFile = options.cypressConfig.indexHtmlFile
 
   let specsPathsSet = getSpecsPathsSet(specs)
-  // TODO: use async fs methods here
-  // eslint-disable-next-line no-restricted-syntax
-  let loader = fs.readFileSync(INIT_FILEPATH, 'utf8')
+
+  // Load the init file asynchronously with proper error handling
+  const loadInitFile = async (): Promise<string> => {
```

Comment:
@lucgonp Do you have any performance benchmarks to share to show the improved performance here? I feel like there needs to be a tangible benefit here to measure given the increase in code complexity.

Is there a reason we need to have the promise be a singleton. I'm guessing there is some type of race condition when loading the file?